indico/modules/events/contributions/models/contributions.py

```diff
@@ -510,7 +511,19 @@ def enabled_editables(self):
             return []
 
         enabled_editable_types = editing_settings.get(self.event, 'editable_types')
-        return [editable for editable in self.editables if editable.type.name in enabled_editable_types]
+        enabled_editables = [editable for editable in self.editables if editable.type.name in enabled_editable_types]
+        order = list(EditableType)
+        return sorted(enabled_editables, key=lambda editable: order.index(editable.type))
```

Comment:
This is a quite interesting way of doing the sorting, but I would probably find something like this more readable:

```python
available_editable_types = {editable.type.name for editable in self.editables}
return [t for t in EditableType if t.name in (available_editable_types & enabled_editable_types)]
```
(untested, though)

Reply:
but this returns the actual editables, no the types

Reply:
> but this returns the actual editables, not the types

Indeed. Never mind 🤦